backend/core/domains/payments/services/invoice_service.py

```
@@ -201,15 +212,476 @@ def create_from_quote(quote, due_days=14):
+            # Validate invoice status
+            if invoice.status != 'ISSUED':
+                raise ValueError(f'Cannot create payment plan for invoice with status {invoice.get_status_display()}')
+
+            # Check if payment plan already exists
+            if hasattr(invoice.event, 'payment_plan'):
+                raise ValueError('A payment plan already exists for this event')
+
+            # Validate down payment amount doesn't exceed total
+            down_payment = plan_data.get('down_payment_amount', 0)
+            if down_payment >= invoice.total_amount:
+                raise ValueError('Down payment amount cannot exceed invoice total')
```

Ruff (0.13.3)

Abstract `raise` to an inner function

(TRY301)